delay/delay.go

```diff
@@ -98,6 +104,43 @@ func isContext(t reflect.Type) bool {
 	return t == stdContextType || t == netContextType
 }
 
+// fileKey finds a stable representation of the caller's file path.
+// For calls from package main: strip all leading path entries, leaving just the filename.
+// For calls from anywhere else, strip $GOPATH/src, leaving just the package path and file path.
+func fileKey(file string) (string, error) {
+	if !internal.IsSecondGen() {
+		return file, nil
+	}
+	// If the caller is in the same Dir as mainPath, then strip everything but the file name.
+	if filepath.Dir(file) == appengine.MainPath {
```

Comment:
Trying to poke holes here: if `appengine.MainPath` isn't set, then this condition will never match, since according to the docs for [`filepath.Dir`](https://godoc.org/path/filepath#Dir):

> If the path is empty, Dir returns ".".

Reply:
If it's unset, then we're on GAE first-gen which doesn't need this logic, and which would never get past the `!internal.IsSecondGen()` check.

Reply:
That's reasonable, but let's add commentary to inform the reader.

Reply:
I added a check if MainPath == ""